### PR TITLE
Branch file to bucket

### PR DESCRIPTION
**Pull request by mivelRBFA**
Now the log.txt is needed.

### CODE REVIEW

File: log.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+2022-08-19 16:43:16.146940
```
Comment (GiunioRBFA) [inline log.txt]: what is this?

File: main.py [hunk 2]
```diff
@@ -24,3 +25,19 @@ async def read_root():
         return {f"{now} inserted in bigquery table"}
     else:
         return {f"{now} not inserted in bigquery table"}
+
+
+
+@app.get("/file_to_bucket")
+async def upload_file():
+    now = datetime.now()
+    f = open("log.txt", "a")
```
Comment (GiunioRBFA) [inline main.py]: you don't need to write a new file locally, leverage the tempfile library to create a temporary file and upload it to the bucket

Comment (GiunioRBFA) [reply main.py]: You can also use the StringIO module

File: main.py [hunk 3]
```diff
@@ -24,3 +25,19 @@ async def read_root():
...
+    f.write(f"{now}" + "\n")
+    f.close()
+    storage_client = storage.Client()
+    my_bucket = storage_client.get_bucket("rbfa-workshop-sandboxes-milanvelle")
+    blob = my_bucket.blob("Timestamps/" + f"{now}")
+    file_path = r"C:\Users\mivel.ext\Documents\PyCharm_Projecten\TestProject\log.txt"
```
Comment (GiunioRBFA) [inline main.py]: this won't work on a Unix machine (Linux or Mac). Leverage the pathlib module to make sure that your path are cross-OS compatible

File: log.txt (hunk 1 above)
Comment (mivelRBFA) [reply log.txt]: I upload this file to the bucket folder with also the new timestamp as its name

File: main.py [hunk 4]
```diff
@@ -1,10 +1,9 @@
-from fastapi import FastAPI
 from datetime import datetime
-from google.cloud import bigquery
-
+from fastapi import FastAPI
+from google.cloud import bigquery, storage
+import tempfile
```
Comment (GiunioRBFA) [inline main.py]: did you run the hooks?

File: main.py [hunk 5]
```diff
@@ -24,3 +23,16 @@ async def read_root():
...
+
+
+@app.get("/file_to_bucket")
+async def upload_file():
+    now = datetime.now()
+    storage_client = storage.Client()
+    my_bucket = storage_client.get_bucket("rbfa-workshop-sandboxes-milanvelle")
+    blob = my_bucket.blob("Timestamps/" + f"{now}")
+    tf=tempfile.NamedTemporaryFile(mode='w+b',suffix='.csv', prefix=now.strftime('%d-%m-%y--%H-%M-%S'),delete=False)
+    tf.close()
```
Comment (GiunioRBFA) [inline main.py:43]: nice!

File: requirements.txt [hunk 6]
```diff
@@ -1,6 +1,8 @@
 FastAPI == 0.79.1
 gunicorn == 20.1.0
-uvicorn==0.18.3
+uvicorn == 0.18.3
 BigQuery-Python == 1.15.0
```
Comment (GiunioRBFA) [inline requirements.txt]: I am not whether this Bigquery-Python package is required

File: log.txt [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+2022-08-24 15:01:00.101621
```
Comment (GiunioRBFA) [inline log.txt]: this file must be deleted!

**Pull request merged**